src/Compilers/VisualBasic/Portable/Parser/ParseExpression.vb

```diff
@@ -1399,60 +1399,70 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
             Do
                 ' Check for the first keyword argument.
 
+                Dim isNamed As Boolean = False
                 If (CurrentToken.Kind = SyntaxKind.IdentifierToken OrElse CurrentToken.IsKeyword()) AndAlso
                     PeekToken(1).Kind = SyntaxKind.ColonEqualsToken Then
 
-                    ParseNamedArguments(arguments)
-                    Exit Do
+                    If _scanner.Options.LanguageVersion.AllowNonTrailingNamedArguments() Then
```

Comment:
>If _scanner.Options.LanguageVersion.AllowNonTrailingNamedArguments() Then [](start = 20, length = 73)

I think this logic is actually going to produce trees of different shape depending on the language version. We probably do not want that. Consider getting rid of the ParseNamedArguments helper completely and do all the work here.